Code/ForceField/ANI/AtomicContrib.cpp

```diff
@@ -0,0 +1,224 @@
+                               unsigned int numAtoms, unsigned int numLayers,
+                               unsigned int ensembleSize,
+                               std::string modelType) {
+  PRECONDITION(owner, "Bad Owner")
+  PRECONDITION(atomType == 0 || atomType == 1 || atomType == 2 || atomType == 3,
+               "Atom Type not Supported");
+  PRECONDITION(modelType == "ANI-1x" || modelType == "ANI-1ccx",
+               "Model Not currently supported")
+  PRECONDITION(ensembleSize > 0,
+               "There must be at least 1 model in the ensemble");
+  URANGE_CHECK(atomIdx, numAtoms);
+  dp_forceField = owner;
+  this->d_atomType = atomType;
+  this->d_atomIdx = atomIdx;
+  this->d_speciesVec = speciesVec;
+  this->d_numAtoms = numAtoms;
+  this->d_ensembleSize = ensembleSize;
+  this->d_modelType = modelType;
+
+  if (this->d_atomEncoding.find(this->d_atomType) !=
+      this->d_atomEncoding.end()) {
+    auto atomicSymbol = this->d_atomEncoding[this->d_atomType];
+    for (unsigned int modelNum = 0; modelNum < ensembleSize; modelNum++) {
+      std::vector<ArrayXXd> currModelWeights;
+      std::vector<ArrayXXd> currModelBiases;
+      Utils::loadFromBin(&currModelWeights, &currModelBiases, modelNum,
+                         atomicSymbol, this->d_modelType);
+      this->d_weights.push_back(currModelWeights);
+      this->d_biases.push_back(currModelBiases);
+    }
+    Utils::loadSelfEnergy(&(this->d_selfEnergy), atomicSymbol,
+                          this->d_modelType);
+  } else {
+    this->d_selfEnergy = 0;
+  }
+}
+
+double ANIAtomContrib::forwardProp(ArrayXXd aev) const {
+  if (this->d_atomType == -1) {
+    return 0;
+  }
+  if (aev.cols() != 1) {
+    aev.transposeInPlace();
+  }
+
+  std::vector<double> energies;
+
+  for (unsigned int modelNo = 0; modelNo < this->d_weights.size(); modelNo++) {
+    auto temp = aev;
+    for (unsigned int layer = 0; layer < this->d_weights[modelNo].size();
+         layer++) {
+      temp =
+          ((this->d_weights[modelNo][layer].matrix() * temp.matrix()).array() +
+           this->d_biases[modelNo][layer])
+              .eval();
+      if (layer < this->d_weights[modelNo].size() - 1) Utils::CELU(temp, 0.1);
+    }
+    energies.push_back(temp.coeff(0, 0));
+  }
+  return std::accumulate(energies.begin(), energies.end(), 0.0) /
+         energies.size();
+}
+
+double ANIAtomContrib::getEnergy(double *pos) const {
+  auto aev = RDKit::Descriptors::ANI::AtomicEnvironmentVector(
+      pos, this->d_speciesVec, this->d_numAtoms);
+  return this->ANIAtomContrib::forwardProp(aev.row(this->d_atomIdx)) +
+         this->d_selfEnergy;
+}
+
+double ANIAtomContrib::getEnergy(Eigen::ArrayXXd aev) const {
+  return this->ANIAtomContrib::forwardProp(aev.row(this->d_atomIdx)) +
+         this->d_selfEnergy;
+}
+
+void ANIAtomContrib::getGrad(double *pos, double *grad) const {}
+
+namespace Utils {
+
+double RELU(double val) { return std::max((double)0, val); }
```

Comment:
Nit: you can just use `0.0` and the C++ standard says that will be a double:
```suggestion
double RELU(double val) { return std::max(0.0, val); }
```